### PR TITLE
Fixed show search tabs string and tab search pref not being read on initial start

### PR DESCRIPTION
**Pull request by nullhook**
Resolves https://github.com/brave/brave-browser/issues/17310 and
Resolves https://github.com/brave/brave-browser/issues/17343

Screenshot:
image

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Comment by karenkliu** [comment]
Unassigning myself and assigning @rmcfadden3 since this is purely a text thing.

Btw, the toggle should be default on.

**Comment by rmcfadden3** [comment]
Copy looks good.

File: browser/ui/views/frame/brave_browser_view.cc [hunk 1]
```diff
@@ -128,6 +128,8 @@ BraveBrowserView::BraveBrowserView(std::unique_ptr<Browser> browser)
       kTabsSearchShow,
       base::BindRepeating(&BraveBrowserView::OnPreferenceChanged,
                           base::Unretained(this)));
+  // Show the correct value in settings on initial start
+  BraveBrowserView::OnPreferenceChanged(kTabsSearchShow);
```
Comment (simonhong) [inline browser/ui/views/frame/brave_browser_view.cc:132]: How about creating `UpdateSearchTabButtonState()` and move `OnPreferenceChanged`'s impl to it?
Then, calling `UpdateSearchTabButtonState()` from here and from `OnPreferenceChanged()` ?

**Comment by simonhong** [comment]
and How about adding browser test case that check whether button has proper visibility after startup and whether visibility is matched after changing prefs?

File: browser/ui/views/frame/brave_browser_view.cc [hunk 2]
```diff
@@ -129,7 +129,7 @@ BraveBrowserView::BraveBrowserView(std::unique_ptr<Browser> browser)
       base::BindRepeating(&BraveBrowserView::OnPreferenceChanged,
                           base::Unretained(this)));
   // Show the correct value in settings on initial start
-  BraveBrowserView::OnPreferenceChanged(kTabsSearchShow);
+  BraveBrowserView::UpdateSearchTabsButtonState();
```
Comment (simonhong) [inline browser/ui/views/frame/brave_browser_view.cc:132]: nit: Don't need to call with class name.

File: browser/ui/views/frame/brave_browser_view.cc [hunk 3]
```diff
@@ -164,9 +164,15 @@ BraveBrowserView::BraveBrowserView(std::unique_ptr<Browser> browser)
 
 void BraveBrowserView::OnPreferenceChanged(const std::string& pref_name) {
   if (pref_name == kTabsSearchShow) {
-    auto show_tabs_search = GetProfile()->GetPrefs()->GetBoolean(pref_name);
-    tab_strip_region_view()->tab_search_button()->SetVisible(show_tabs_search);
-    return;
+    BraveBrowserView::UpdateSearchTabsButtonState();
```
Comment (simonhong) [inline browser/ui/views/frame/brave_browser_view.cc:167]: ditto.

**Review by simonhong — APPROVED** [review]
++ 👍🏼

File: app/brave_generated_resources.grd [hunk 4]
```diff
@@ -296,7 +296,7 @@ By installing this extension, you are agreeing to the Google Widevine Terms of U
         Show Brave suggested sites in autocomplete suggestions
       </message>
       <message name="IDS_SETTINGS_TABS_SEARCH_SHOW" desc="The button for searching tabs on the left of tab strip region.">
-        Show Search Tabs Button
+        Show Search tabs button
```
Comment (petemill) [inline app/brave_generated_resources.grd:299]: "Show tab search button"

**Review by petemill — CHANGES_REQUESTED** [review]
minor string change

**Review by simonhong — APPROVED** [review]
++

**Review by petemill — APPROVED** [review]
beautiful test!

**Pull request merged**